yarn-project/simulator/src/avm/avm_memory_types.ts

```diff
@@ -382,21 +383,23 @@ export class TaggedMemory implements TaggedMemoryInterface {
   public static getTag(v: MemoryValue | undefined): TypeTag {
     let tag = TypeTag.INVALID;
 
+    // Not sure why, but using instanceof here doesn't work and leads odd behavior,
+    // but using constructor.name does the job...
     if (v === undefined) {
       tag = TypeTag.FIELD; // uninitialized memory is Field(0)
-    } else if (v instanceof Field) {
+    } else if (v.constructor.name == 'Field') {
       tag = TypeTag.FIELD;
-    } else if (v instanceof Uint1) {
+    } else if (v.constructor.name == 'Uint1') {
       tag = TypeTag.UINT1;
-    } else if (v instanceof Uint8) {
+    } else if (v.constructor.name == 'Uint8') {
       tag = TypeTag.UINT8;
-    } else if (v instanceof Uint16) {
+    } else if (v.constructor.name == 'Uint16') {
       tag = TypeTag.UINT16;
-    } else if (v instanceof Uint32) {
+    } else if (v.constructor.name == 'Uint32') {
       tag = TypeTag.UINT32;
-    } else if (v instanceof Uint64) {
+    } else if (v.constructor.name == 'Uint64') {
       tag = TypeTag.UINT64;
-    } else if (v instanceof Uint128) {
+    } else if (v.constructor.name == 'Uint128') {
       tag = TypeTag.UINT128;
     }
```

Comment:
This is silly.... I lost like 2 hours debugging an error with tag checking until eventually I resorted to this change. I was printing `v` right before this if/else and it was showing `Uint32(0x3)` for example, but would then fail `instanceof Uint32` and I don't understand why. I saw it also for Field and the others. @fcarreiro